### PR TITLE
Test LPC with both float32 and float64

### PR DESCRIPTION
**Pull request by ajweiss**
#### Reference Issue

Adds tests for #856

#### What does this implement/fix? Explain your changes.

Adds explicit tests for single and double precision numbers for `librosa.lpc`

#### Any other comments?

I wasn't able to reproduce the issue by testing locally against Python 2.7 and 3.7.2.  (with numba 0.40.1 and 0.43.1 respectively).  Opening this PR because the tests are useful regardless and maybe the issue will surface in the CI.

### CODE REVIEW

**Comment by bmcfee** [comment]
Very strange, it definitely reproduces on my machine with a 3.6 env.  It also fails for me in a 3.7 environment with numba 0.40.1.

**Comment by bmcfee** [comment]
Picking this one up again -- I pulled down your branch, and the tests pass on my 3.6 conda environment.  However, the following still happens:
```python
In [15]: y = np.random.randn(sr).astype(np.float32)                                           

In [16]: a = librosa.lpc(y, 5)                                                                
---------------------------------------------------------------------------
TypingError                               Traceback (most recent call last)
<ipython-input-16-563b53ac7d1d> in <module>
----> 1 a = librosa.lpc(y, 5)

~/git/librosa/librosa/core/audio.py in lpc(y, order)
    586     util.valid_audio(y, mono=True)
    587 
--> 588     return __lpc(y, order)
    589 
    590 

~/miniconda3/envs/py36/envs/py36/envs/py36/envs/py36/envs/py36/lib/python3.6/site-packages/numba/dispatcher.py in _compile_for_args(self, *args, **kws)
    348                 e.patch_message(msg)
    349 
--> 350             error_rewrite(e, 'typing')
    351         except errors.UnsupportedError as e:
    352             # Something unsupported is present in the user code, add help info

~/miniconda3/envs/py36/envs/py36/envs/py36/envs/py36/envs/py36/lib/python3.6/site-packages/numba/dispatcher.py in error_rewrite(e, issue_type)
    315                 raise e
    316             else:
--> 317                 reraise(type(e), e, None)
    318 
    319         argtypes = []

~/miniconda3/envs/py36/envs/py36/envs/py36/envs/py36/envs/py36/lib/python3.6/site-packages/numba/six.py in reraise(tp, value, tb)
    656             value = tp()
    657         if value.__traceback__ is not tb:
--> 658             raise value.with_traceback(tb)
    659         raise value
    660 

TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Cannot unify array(float32, 1d, C) and array(float64, 1d, C) for 'fwd_pred_error', defined at /home/bmcfee/git/librosa/librosa/core/audio.py (612)

File "git/librosa/librosa/core/audio.py", line 612:
def __lpc(y, order):
    <source elided>
    # iteration of the below loop
    fwd_pred_error = y[1:]
    ^

[1] During: typing of assignment at /home/bmcfee/git/librosa/librosa/core/audio.py (645)

File "git/librosa/librosa/core/audio.py", line 645:
def __lpc(y, order):
    <source elided>
        fwd_pred_error_tmp = fwd_pred_error
        fwd_pred_error = fwd_pred_error + reflect_coeff*bwd_pred_error
        ^

This is not usually a problem with Numba itself but instead often caused by
the use of unsupported features or an issue in resolving types.

To see Python/NumPy features supported by the latest release of Numba visit:
http://numba.pydata.org/numba-doc/dev/reference/pysupported.html
and
http://numba.pydata.org/numba-doc/dev/reference/numpysupported.html

For more information about typing errors and how to debug them visit:
http://numba.pydata.org/numba-doc/latest/user/troubleshoot.html#my-code-doesn-t-compile

If you think your code should work with Numba, please report the error message
and traceback, along with a minimal reproducer at:
https://github.com/numba/numba/issues/new
```

More to follow in CR comments.

File: tests/test_core.py [hunk 1]
```diff
@@ -627,16 +627,18 @@ def __test(signal, order, true_coeffs, est_coeffs):
...
     n = 5000
-    est_a = np.zeros((n, 6))
-    truth_a = [1, 0.5, 0.4, 0.3, 0.2, 0.1]
+    est_a = np.zeros((n, 6), dtype=dtype)
+    truth_a = np.array([1, 0.5, 0.4, 0.3, 0.2, 0.1], dtype=dtype)
     for i in range(n):
-        noise = np.random.randn(1000)
+        noise = np.random.randn(1000).astype(dtype)
         filtered = scipy.signal.lfilter([1], truth_a, noise)
         est_a[i, :] = librosa.lpc(filtered, 5)
```
Comment (bmcfee) [inline tests/test_core.py]: `lfilter` is returning `filtered` as having `dtype=np.float64`, and this test is passing.

Changing the lpc call to `librosa.lpc(filtered.astype(dtype), 5)` exposes the type error originally reported.

**Comment by bmcfee** [comment]
Hi @ajweiss -- do you still have some time to look into this?

**Comment by ajweiss** [comment]
Hey, sorry for delay... I'll pick this up this week.

**Comment by ajweiss-dd** [comment]
This should fix it... not sure if I like this style of casting things, but I'm not sure what else I can do...

**Comment by bmcfee** [comment]
Thanks @ajweiss ! I took the liberty of resolving some merge conflicts since this is now out of sync with master.  Hopefully this gets it all resolved and we can close it out.

**Comment by bmcfee** [comment]
Ok, this looks good now.  Thanks @ajweiss !  :shipit:

**Pull request merged**